src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoUnitTest.java

```diff
@@ -0,0 +1,749 @@
+    private static final String DATASET_PROJECT_ID = "dataset_data";
+    private static final Instant SNAPSHOT_CREATION = Instant.now();
+    private static final String DATASET_NAME = "datasetName";
+    private static final String SNAPSHOT_DESCRIPTION = "snapshotDescription";
+    private static final String TABLE_1_NAME = "tableA";
+    private static final UUID TABLE_1_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_1_ID = UUID.randomUUID();
+    private static final String TABLE_1_COL1_NAME = "col1a";
+    private static final UUID TABLE_1_COL1_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_1_COL1_ID = UUID.randomUUID();
+    private static final String TABLE_1_COL2_NAME = "col2a";
+    private static final UUID TABLE_1_COL2_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_1_COL2_ID = UUID.randomUUID();
+
+    private static final String TABLE_2_NAME = "tableB";
+    private static final UUID TABLE_2_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_2_ID = UUID.randomUUID();
+    private static final String TABLE_2_COL1_NAME = "col1b";
+    private static final UUID TABLE_2_COL1_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_2_COL1_ID = UUID.randomUUID();
+    private static final String TABLE_2_COL2_NAME = "col2b";
+    private static final UUID TABLE_2_COL2_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_2_COL2_ID = UUID.randomUUID();
+    private static final String TABLE_2_COL3_NAME = "col3b";
+    private static final UUID TABLE_2_COL3_ID = UUID.randomUUID();
+    private static final UUID SNAPSHOT_TABLE_2_COL3_ID = UUID.randomUUID();
+
+    @Mock
+    private ApplicationConfiguration applicationConfiguration;
+    @Mock
+    private BigQueryConfiguration bigQueryConfiguration;
+    @Mock
+    private BigQueryProject bigQueryProjectSnapshot;
+    @Mock
+    private BigQuery bigQuerySnapshot;
+    @Mock
+    private BigQueryProject bigQueryProjectDataset;
+    @Mock
+    private BigQuery bigQueryDataset;
+    @Mock
+    private BigQueryProjectProvider bigQueryProjectProvider;
+
+    private Snapshot snapshot;
+    private BigQueryPdao dao;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        when(bigQueryProjectSnapshot.getProjectId()).thenReturn(SNAPSHOT_PROJECT_ID);
+        when(bigQueryProjectSnapshot.getBigQuery()).thenReturn(bigQuerySnapshot);
+        when(bigQueryProjectProvider.apply(SNAPSHOT_PROJECT_ID)).thenReturn(bigQueryProjectSnapshot);
+
+        when(bigQueryProjectDataset.getProjectId()).thenReturn(DATASET_PROJECT_ID);
+        when(bigQueryProjectDataset.getBigQuery()).thenReturn(bigQueryDataset);
+        when(bigQueryProjectProvider.apply(DATASET_PROJECT_ID)).thenReturn(bigQueryProjectDataset);
+
+        dao = new BigQueryPdao(applicationConfiguration, bigQueryConfiguration, bigQueryProjectProvider);
```

Comment:
For cases like `applicationConfiguration` and `bigQueryConfiguration`, where the class is mocked but the mock is never used in the test code, I find it clearer to create the mock directly instead of using a field in the class. Fewer fields in the class make it easier to see what state is important and what state isn't.

Reply:
Fair enough!